plugins/modules/proxmox_ipam_info.py

```diff
@@ -0,0 +1,236 @@
+                "zone": "test1"
+            },
+            {
+                "hostname": "ns4.proxmox.pc",
+                "ip": "10.10.0.7",
+                "mac": "BC:24:11:D5:CD:82",
+                "subnet": "10.10.0.0/24",
+                "vmid": 103,
+                "vnet": "test2",
+                "zone": "test1"
+            },
+            {
+                "gateway": 1,
+                "ip": "10.10.0.1",
+                "subnet": "10.10.0.0/24",
+                "vnet": "test2",
+                "zone": "test1"
+            },
+            {
+                "hostname": "ns2.proxmox.pc.test3",
+                "ip": "10.10.0.5",
+                "mac": "BC:24:11:86:77:56",
+                "subnet": "10.10.0.0/24",
+                "vmid": 101,
+                "vnet": "test2",
+                "zone": "test1"
+            }
+        ]
+    }
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.proxmox.plugins.module_utils.proxmox import (
+    proxmox_auth_argument_spec,
+    ProxmoxAnsible
+)
+
+
+def get_proxmox_args():
+    return dict(
+        ipam=dict(type="str", required=False),
+        vmid=dict(type='int', required=False)
+    )
+
+
+def get_ansible_module():
+    module_args = proxmox_auth_argument_spec()
+    module_args.update(get_proxmox_args())
+    return AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+
+
+class ProxmoxIpamInfoAnsible(ProxmoxAnsible):
+    def __init__(self, module):
+        super(ProxmoxIpamInfoAnsible, self).__init__(module)
+        self.params = module.params
+
+    def run(self):
+        vmid = self.params.get('vmid')
+        ipam = self.params.get('ipam')
+        if vmid:
+            self.module.exit_json(
+                changed=False, ips=self.get_ip_by_vmid(vmid)
+            )
+
+        elif self.params.get('ipam'):
+            if ipam not in self.get_ipams():
+                self.module.fail_json(
+                    msg=f'Ipam {ipam} is not present'
+                )
+            else:
+                self.module.exit_json(
+                    changed=False,
+                    ipams=self.get_ipam_status()[ipam]
+                )
+        else:
+            self.module.exit_json(
+                changed=False,
+                ipams=self.get_ipam_status()
+            )
+
+    def get_ipams(self):
+        try:
+            ipams = self.proxmox_api.cluster().sdn().ipams().get()
+            return [ipam['ipam'] for ipam in ipams]
+        except Exception as e:
+            self.module.fail_json(
+                msg=f'Failed to retrieve ipam information {e}'
+            )
+
+    def get_ipam_status(self):
+        try:
+            ipam_status = dict()
+            ipams = self.get_ipams()
+            for ipam_id in ipams:
+                ipam = getattr(self.proxmox_api.cluster().sdn().ipams(), ipam_id)
+                ipam_status[ipam_id] = ipam().status().get()
```

Comment:
I, personally, think this is more concise.

```suggestion
                ipam_status[ipam_id] = self.proxmox_api.cluster().sdn().ipams(ipam_id).status().get()
```